### PR TITLE
Add 1.0.4 commands to encoding

### PR DESCRIPTION
**Pull request by lucasgranberg**
Adding the commands makes it easier to develop towards 1.0.4. As I added them here even 1.0.2 projects would have the new commands. Ideally these new commands could be behind a feature but that is not so easy to implement with the current macros.

### CODE REVIEW

**Review by lulf — APPROVED** [review]
LGTM!

File: encoding/tests/maccommandcreator.rs [hunk 1]
```diff
@@ -152,18 +152,52 @@ fn test_rx_timing_setup_req_creator() {
     let res = creator.set_delay(0x0f).unwrap().build();
     assert_eq!(res, [RXTimingSetupReqPayload::cid(), 0x0f]);
 }
-
+#[test]
```
Comment (ivajloip) [inline encoding/tests/maccommandcreator.rs:155]: nit: New line missing here and below.

**Review by ivajloip — CHANGES_REQUESTED** [review]
Thank for implementing those newer mac commands! This is really appreciated! I added a few things to double check, but otherwise it looks good to me! :slightly_smiling_face:

File: encoding/src/maccommandcreator.rs [hunk 2]
```diff
@@ -604,6 +604,112 @@ pub struct RXTimingSetupAnsCreator {}
...
+    }
+    pub fn set_max_eirp(&mut self, max_eirp: u8) -> Result<&mut Self, &str> {
+        if max_eirp > 0x07 {
+            return Err("max_eirp out of range");
+        }
+        self.data[1] &= 0xf8;
+        self.data[1] |= max_eirp;
+
+        Ok(self)
+    }
```
[the remark this answers is not in the document]
Comment (lucasgranberg) [reply encoding/src/maccommandcreator.rs:632]: yep you are right, I read that wrong, max_eirp is 4 bits 3:0 according to the spec

File: encoding/src/maccommands.rs [hunk 3]
```diff
@@ -809,3 +864,75 @@ impl<'a> RXTimingSetupReqPayload<'a> {
         self.0[0] & 0x0f
     }
 }
+
+impl<'a> TXParamSetupReqPayload<'a> {
+    pub fn downlink_dwell_time(&self) -> bool {
+        self.0[0] & (1 << 4) != 0
+    }
+    pub fn uplink_dwell_time(&self) -> bool {
+        self.0[0] & (1 << 3) != 0
```
[the remark this answers is not in the document]
Comment (lucasgranberg) [reply encoding/src/maccommands.rs:873]: yep, brain fart continues from max eirp

**Review by ivajloip — APPROVED** [review]
Thanks! :slightly_smiling_face:

**Pull request merged**